kernel/kernel-api/src/main/java/io/delta/kernel/types/MixedDataType.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.types;
+
+import io.delta.kernel.data.ColumnVector;
+
+/**
+ * TODO:
+ * This is a controversial data type to have, but we have no way to specify the schema
+ * of JSON serialized table schema. In order to use the
+ * {@link io.delta.kernel.client.JsonHandler#parseJson(ColumnVector, StructType)}, the Kernel
+ * needs to specify the exact schema of the data, but depending upon the type serialized (`int` vs
+ * `map`), the serialized JSON could have different schema.
+ * <p>
+ * `int` type column schema is serialized as:
+ *
+ *  <pre>
+ *   {
+ *     "name" : "a",
+ *     "type" : "integer",
+ *     "nullable" : false,
+ *     "metadata" : { }
+ *   }
+ *  </pre>
+ * <p>
+ * `struct` type column schema is serialized as:
+ * <pre>
+ *   {
+ *     "type" : "struct",
+ *     "fields" : [ {
+ *       "name" : "d",
+ *       "type" : "integer",
+ *       "nullable" : false,
+ *       "metadata" : { }
+ *     } ]
+ *   }
+ * </pre>
+ * Reader implementations should convert interpret the values encountered either as `string`
```

Comment:
Can we just clarify that it should either be the raw string value encountered, or the _JSON string_ of the struct type?